### PR TITLE
Feature/add trip

### PR DESCRIPTION
**Pull request by DaveKrauss**
**Pull Request for User Story 3 - Create a trip**

## Acceptance criteria

- [x] Clicking on the `+ add trip` link on the homepage leads to a `form` page
- [x] The `form` page consists of the following:
    - [x] Heading of My Trips
    - [x] Form
        - [x] Title of the Trip: The field is required, and can contain a maximum of 30 characters
        - [x] Starting date: The field is required, with a datepicker that fills the field after clicking on a date
        - [x] Ending date: The field is required, with a datepicker that fills the field after clicking on a date
        - [x] City: The field is required, and can contain a maximum of 30 characters
        - [x] Country: The field is required, and can contain a maximum of 30 characters
        - [x] Image (Url): The field is optional, without a maximum of characters
        - [x] Description: The field is optional, with a maximum of 150 characters
    - [x] An `add trip` button
    - [x] A `cancel` link
    - [x] A `back-link` arrow

_Note: When the maximum amount of characters for a form field hits its treshold, no new characters inputs are added._

- [x] Clicking on the `cancel` link or the `back-link` will open a message below the form, which asks for the users next process steps either to continue or cancel adding a new trip (we might remove the feature on the backlink later, because it might not be handy for the user)
 - [x] If the user exits the clearance process by clicking on No*, the message will disappear, the user can edit the last status of the form  (*we changed no to "continue")
 - [x] If the user exits the clearance process by clicking on Yes, the user will be lead to the homepage

- [x] Clicking on the `add trip` button with all the required fields supplied will send the user to a new page, where the message: "Congratulations, you have a added a new trip" will be displayed. On this page there is a `home` link, that leads the user back to the homepage. This trip is now added to the list of trips on the homepage 
- [x] Clicking on the `add trip` button while at least one required field is not supplied will inform the user about missing entries

_Note: Considering the wireframe `first new trip` , this task will be implemented with US4 (delete a trip). Otherwise it wouldn't be logical at this state._

### CODE REVIEW

File: components/TripForm/index.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+            cols="30"
+            maxLength="150"
+            id="description"
+            name="description"
+            type="text"
+            required
+            placeholder="Enter your description"
+          ></textarea>
+          <p>
+            <span>150</span> characters left
```
Comment (TobiasSteinhagen) [inline components/TripForm/index.js:91]: Why do you use the span for the amount of characters?  🤔

File: components/TripForm/index.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+            required
+          ></input>
+
+          <label htmlFor="startDate">Starting date (dd/mm/yyyy)*</label>
+          <input id="startDate" name="startDate" type="date" required></input>
+
+          <label htmlFor="endDate">Ending date (dd/mm/yyyy)*</label>
+          <input id="endDate" name="endDate" type="date" required></input>
+
+          <label htmlFor="city">City (max. 30 characters)*</label>
```
Comment (TobiasSteinhagen) [inline components/TripForm/index.js:50]: Do you need the htmlFor attribute in react? We don't have but it works totally fine.

Could you explain why you include this attribute? 😊

**Review by TobiasSteinhagen — APPROVED** [review]
LGTM overall, just a few questions and suggestions 🐸🚀

File: components/TripForm/index.js (hunk 2 above)
Comment (DaveKrauss) [reply components/TripForm/index.js:50]: htmlFor links a label and the corresponding input in react. If you click on the label of an input field, you will be directed into the input field.

File: pages/addTrip.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import TripForm from "@/components/TripForm";
+import Link from "next/link";
```
[the remark this answers is not in the document]
Comment (DaveKrauss) [reply pages/addTrip.js]: Yeah, thx for reminding. We removed it 👍

File: pages/index.js [hunk 4]
```diff
@@ -1,10 +1,13 @@
 import TripList from "@/components/TripList";
+import TripForm from "@/components/TripForm";
```
[the remark this answers is not in the document]
Comment (DaveKrauss) [reply pages/index.js]: We used it earlier while testing, and forgot to remove it ^^

File: components/TripForm/index.js (hunk 1 above)
Comment (DaveKrauss) [reply components/TripForm/index.js]: removed, thx

**Pull request merged**